memo/managers.py

```diff
@@ -73,3 +74,63 @@ def get_card_amount_in_deck(self, deck):
 class DeckManager(models.Manager):
     def get_decks_by_owner(self, user):
         return self.filter(owner=user)
+
+
+class CardAnswerHistoryManager(models.Manager):
+    def get_consecutive_days(self, user):
+        from .models import Card
+
+        max_daily_date_times = (
+            self.filter(owner=user)
+            .extra(select={"day": "date( created_at )"})
+            .values("day")
+            .annotate(latest_datetime=Max("created_at"))
+        )
+
+        objects = (
+            self.filter(
+                created_at__in=[
+                    entry["latest_datetime"] for entry in max_daily_date_times
+                ]
+            )
+            .values("card", "created_at")
+            .order_by("-created_at")
+        )
+
+        result = []
+
+        if objects:
+            answered_today = False
+            today = datetime.today()
+            count = 0
+            one_day_less = 1
+
+            for index in range(0, len(objects)):
+                one_day_before = (today - timedelta(days=one_day_less)).strftime(
+                    "%Y-%m-%d"
+                )
+                created_at = (objects[index].get("created_at")).strftime("%Y-%m-%d")
```

Comment:
```suggestion
                created_at = object.created_at.strftime("%Y-%m-%d")
```

Reply:
The `object` is actually a dict, it doesn't allow me to access the data with .

```
>>> print(type(object))
<class 'dict'>
>>> print(dir(object))
['__class__', '__class_getitem__', '__contains__', '__delattr__', '__delitem__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__ior__', '__iter__', '__le__', '__len__', '__lt__', '__ne__', '__new__', '__or__', '__reduce__', '__reduce_ex__', '__repr__', '__reversed__', '__ror__', '__setattr__', '__setitem__', '__sizeof__', '__str__', '__subclasshook__', 'clear', 'copy', 'fromkeys', 'get', 'items', 'keys', 'pop', 'popitem', 'setdefault', 'update', 'values']
```

Should we change the variable name? :thinking:

Reply:
It becomes a class object (instead of a dict) if you remove the `.values` method from your queryset, but if it messes with the query result, it's ok to leave it.